src/il_representations/scripts/dqn_train.py

```diff
@@ -0,0 +1,195 @@
+                         freeze_pol_encoder=freeze_encoder,
+                         policy_class=CnnPolicy,
+                         encoder_path=encoder_path,
+                         encoder_kwargs=encoder_kwargs,
+                         optimizer_class=optimizer_class,
+                         optimizer_kwargs=optimizer_kwargs,
+                         lr_schedule=lr_schedule).to(device)
+
+    color_space = auto_env.load_color_space()
+    augmenter = augmenter_from_spec(augs, color_space)
+    dataset_length = len(dict_dataset['obs']) if n_trans is None else n_trans
+    progress_df = pd.DataFrame()
+
+    trainer = DQN(
+        policy='CnnPolicy',
+        env=venv_chans_first,
+        device=device_name,
+        buffer_size=dataset_length,
+        batch_size=batch_size,
+        optimize_memory_usage=True
+    )
+    trainer.policy = policy
+
+    # Push data into DQN agent's memory.
+    print('Loading data...')
+    for idx in range(dataset_length):
+        if idx % 1000 == 0:
+            print(f'Loading {idx}/{dataset_length}...')
+        obs, next_obs, action, reward, done = dict_dataset['obs'][idx], \
+                                              dict_dataset['next_obs'][idx], \
+                                              dict_dataset['acts'][idx], \
+                                              dict_dataset['rews'][idx], \
+                                              dict_dataset['dones'][idx]
+        # obs = preprocess_obs(th.tensor(obs),
+        #                      observation_space,
+        #                      normalize_images=True)
+        # next_obs = preprocess_obs(th.tensor(next_obs),
+        #                           observation_space,
+        #                           normalize_images=True)
+        # obs, next_obs = th.unsqueeze(obs, dim=0), th.unsqueeze(next_obs, dim=0)
+        # if augmenter is not None:
+            # Here we unsqueeze the obs first since the augmenter only takes
+            # [N, C, H, W] inputs, so we need to fake a "batch size" here.
+            # obs = augmenter(obs)
+            # next_obs = augmenter(next_obs)
+
+        trainer.replay_buffer.add(obs=obs,
+                                  next_obs=next_obs,
+                                  action=action,
+                                  reward=reward,
+                                  done=done)
+
+    # Call DQN training.
+    n_update_per_epoch = int(n_batches / nominal_num_epochs)
+    model_saver = ModelSaver(trainer.policy,
+                             save_dir=os.path.join(out_dir, 'snapshots'),
+                             save_interval_batches=save_every_n_batches)
+    # trainer.learn(total_timesteps=1000000,
+    #               eval_log_path=os.path.join(out_dir, 'progress.csv'))
+
+
+    for epoch in range(nominal_num_epochs):
+        print(f'Training [{epoch}/{nominal_num_epochs}] epochs...')
+        policy, n_update, loss = trainer.train(n_update_per_epoch,
```

Comment:
Maybe explain that this is just doing gradient updates on the network, and _not_ collecting new rollouts? I wasn't sure whether this was the case initially.

Reply:
Yes it doesn't collect new rollouts. I've added some comments to it.